src/main/java/com/pokegoapi/api/map/pokemon/CatchablePokemon.java

```diff
@@ -151,18 +168,38 @@ public EncounterResult encounterPokemon() throws LoginFailedException,
 			throw new RemoteServerException(e);
 		}
 		encountered = response.getStatus() == EncounterResponse.Status.ENCOUNTER_SUCCESS;
-		return new EncounterResult(response);
+		return new EncounterResult.NormalEncounterResult(response);
+	}
+
+
+	private EncounterResult.DiskEncounterResult diskEncounterPokemon() throws LoginFailedException,
+			RemoteServerException {
+		DiskEncounterMessageOuterClass.DiskEncounterMessage reqMsg = DiskEncounterMessageOuterClass.DiskEncounterMessage
+				.newBuilder().setEncounterId(getEncounterId())
+				.setPlayerLatitude(api.getLatitude())
+				.setPlayerLongitude(api.getLongitude())
+				.setFortId(getSpawnPointId()).build();
+		ServerRequest serverRequest = new ServerRequest(
+				RequestTypeOuterClass.RequestType.DISK_ENCOUNTER, reqMsg);
+		api.getRequestHandler().sendServerRequests(serverRequest);
+		DiskEncounterResponseOuterClass.DiskEncounterResponse response = null;
+		try {
+			response = DiskEncounterResponseOuterClass.DiskEncounterResponse
+					.parseFrom(serverRequest.getData());
+		} catch (InvalidProtocolBufferException e) {
```

Comment:
just for consistency with the rest of API.

also i catch InvalidProtocolBufferException and throw RemoteServerException

take a look to the rest of api, it's done in the same way